x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.deprecation;
+
+import org.elasticsearch.action.admin.cluster.node.info.PluginsAndModules;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.xpack.core.deprecation.DeprecationIssue;
+
+import java.util.Locale;
+
+class NodeDeprecationChecks {
+
+    static DeprecationIssue checkProcessors(final Settings settings , final PluginsAndModules pluginsAndModules) {
+        if (EsExecutors.PROCESSORS_SETTING.exists(settings) == false) {
+            return null;
+        }
+        final String message = String.format(
+            Locale.ROOT,
+            "setting [%s] is deprecated in favor of setting [%s]",
+            EsExecutors.PROCESSORS_SETTING.getKey(),
+            EsExecutors.NODE_PROCESSORS_SETTING.getKey());
+        final String url = "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html#remove-processors";
```

Comment:
This link refers to master, which may be different when someone sees this setting (for example once 7.x is in maintenance mode)? I think we need a better permalink if we are to include this, but haven't we avoided links in exception messages in the past for this exact reason?

Reply:
Let me make a docs push to add a deprecation note to the 7.4 docs which can be used as a permalink.

Reply:
I've just realized we have the same issue for the existing DeprecationIssue instances, which we should probably (separately from this PR) fix.

Reply:
@gwbrown Can you address that?

Reply:
@rjernst I pushed a permalink that is currently broken, but will be fixed by the time the 7.4 release is in published.